main/ejml-core/src/org/ejml/data/ElementLocation.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2020, Peter Abeles. All Rights Reserved.
+ *
+ * This file is part of Efficient Java Matrix Library (EJML).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ejml.data;
+
+/**
+ * The row and column of an element in a Matrix
+ *
+ * @author Peter Abeles
+ */
+public class ElementLocation {
+    /** Row coordinate of an element */
+    public int row;
+
+    /** Column coordinate of an element */
+    public int col;
+
+    public ElementLocation() {}
+
+    public ElementLocation( int row, int col ) {
+        this.row = row;
+        this.col = col;
+    }
+
+    public void set( ElementLocation loc ) {
+        this.row = loc.row;
+        this.col = loc.col;
+    }
+
+    public void set( int row, int col ) {
+        this.row = row;
+        this.col = col;
+    }
+
+    public int getRow() {
+        return row;
+    }
+
+    public void setRow( int row ) {
+        this.row = row;
+    }
+
+    public int getCol() {
+        return col;
+    }
+
+    public void setCol( int col ) {
+        this.col = col;
```

Comment:
why having getters and setters, if the fields are public?
Personally, for this class I would just remove these getters and setters.

Reply:
For serialization and bean support. This came across to me as a data structure someone would try to serialize. I was going to add the Lombok `@Data` annotation and let it handle that mess then I noticed this project doesn't have Lombok. Speaking of Serialization it's still missing boiler plate. Maybe I'll just add Lombok.

Reply:
Makes sense, thank you for reasoning.

I don't have a strong opinion about Lombok. If its the only place, than I am in favour for explicit getters/setters.

Reply:
I think I'm going to add it since there are other places i wanted to make more concise. It's funny, I think until Python's popularity sky rotted a few years ago most people were absolute zealots about encapsulation and would complain that I wasn't using accessors everywhere, even if there was no benefit.